tensorboard/plugins/histogram/histograms_demo.py

```diff
@@ -107,18 +100,17 @@ def run_all(logdir, verbose=False, num_summaries=400):
         "two normal distributions.",
     )
 
-    summaries = tf.compat.v1.summary.merge_all()
 
-    # Setup a session and summary writer
-    sess = tf.compat.v1.Session()
-    writer = tf.summary.FileWriter(logdir)
-
-    # Setup a loop and write the summaries to disk
-    N = num_summaries
-    for step in range(N):
-        k_val = step / float(N)
-        summ = sess.run(summaries, feed_dict={k: k_val})
-        writer.add_summary(summ, global_step=step)
+def run_all(logdir, num_summaries=400):
+    """Generate a bunch of histogram data, and write it to logdir."""
+    tf.random.set_seed(0)
+    writer = tf.summary.create_file_writer(os.path.join(logdir, "my_run_name"))
+    with writer.as_default():
+        for step in range(num_summaries):
+            tf.summary.experimental.set_step(step)
+            k = step / float(num_summaries)
+            run(k)
+            writer.flush()
 
 
 def main(unused_argv):
```

Comment:
Could we add a print statement that informs users about TensorBoard / TensorBoard.dev?

```
print(
        "To view results in your browser, run `tensorboard --logdir %s`"
        % LOGDIR
    )
    print(
        "Logs can be uploaded publicly to TensorBoard.dev via "
        + "`tensorboard dev upload --logdir %s`" % LOGDIR
    )
```

Reply:
Done. For this demo I addd `/my_run_name` to align with above settings

Reply:
Maybe we can just leave off `my_run_name` entirely, both here and in the actual writer construction?  I don't think it's really necessary since we have only a single run here, and omitting it would be more consistent with how this demo worked previously, as well as with the other demos (where it's just `/tmp/scalars_demo` for example).